ibm/service/cos/resource_ibm_cos_bucket.go

```diff
@@ -918,37 +919,40 @@ func resourceIBMCOSBucketUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
-	sess, err := meta.(conns.ClientSession).CosConfigV1API()
-	if err != nil {
-		return err
-	}
+	authenticator := new(core.IamAuthenticator)
+	authenticator.ApiKey = apiKey
+	authenticator.URL = authEndpointPath
+	optionsRC := new(rcsdk.ResourceConfigurationV1Options)
+	optionsRC.Authenticator = authenticator
 	if endpointType == "private" {
-		sess.SetServiceURL("https://config.private.cloud-object-storage.cloud.ibm.com/v1")
+		optionsRC.URL = "https://config.private.cloud-object-storage.cloud.ibm.com/v1"
 	}
 	if endpointType == "direct" {
-		sess.SetServiceURL("https://config.direct.cloud-object-storage.cloud.ibm.com/v1")
+		optionsRC.URL = "https://config.direct.cloud-object-storage.cloud.ibm.com/v1"
 	}
 
 	if apiType == "sl" {
 		satconfig := fmt.Sprintf("https://config.%s.%s.cloud-object-storage.appdomain.cloud/v1", serviceID, bLocation)
 
-		sess.SetServiceURL(satconfig)
+		optionsRC.URL = satconfig
+	}
+	rcClient, clientErr := rcsdk.NewResourceConfigurationV1(optionsRC)
+	if clientErr != nil {
+		return clientErr
 	}
-
 	hasChanged := false
-	updateBucketConfigOptions := &resourceconfigurationv1.UpdateBucketConfigOptions{}
 
 	//BucketName
 	bucketName = d.Get("bucket_name").(string)
-	updateBucketConfigOptions.Bucket = &bucketName
-
+	bucketPatchModel := new(rcsdk.BucketPatch)
 	if d.HasChange("hard_quota") {
 		hasChanged = true
-		updateBucketConfigOptions.HardQuota = aws.Int64(int64(d.Get("hard_quota").(int)))
+		bucketPatchModel.HardQuota = core.Int64Ptr(int64(d.Get("hard_quota").(int)))
+
 	}
 
 	if d.HasChange("allowed_ip") {
-		firewall := &resourceconfigurationv1.Firewall{}
+		firewall := &rcsdk.Firewall{}
```

Comment:
Instead of rcsdk, can we change the variable name here to resourceConfiguration or similar at all the places